### DISCUSSION TITLE
[IMPORTANT] Several breaking issues with the guide

### DISCUSSION THREAD

**Discussion by Dark-Noir** [Guilouz/Klipper-Flsun-Speeder-Pad#187] [Q&A]
!!! FOR ALL USER WHO WANT TO USE THE GUIDE, PLEASE READ THE FOLLOWING !!!

Hey @Guilouz,

i wanted to update my pad (did the old guide), so i tried a clean re-install, but ran into several issues.

<ins>**1.**</ins> After installing the network-manager, it was no longer possible to connect to my WiFi:
<details>
  <summary>Pad display output</summary> 

```
Starting WiFi Re-association  
<3>Trying to associate with b0:f2:08:89:a2:0c (SSID='Router' freq=2412 MHz)  
<3>CTRL-EVENT-DISCONNECTED bssid=b0:f2:08:89:a2:0c reason=1 locally_generated=1  
<3>CTRL-EVENT-REGDOM-CHANGE init=COUNTRY_JE type= COUNTRY alpha2=DE  
<3>CTRL-EVENT-REGDOM-CHANGE init=CORE type= WORLD  
<3>CTRL-EVENT-REGDOM-CHANGE init=CORE type= WORLD  
<3>Trying to associate with b0:f2:08:89:d5:f9 (SSID='Router' freq=2412 MHz)  
<3>Trying to associate with b0:f2:08:89:a2:0c (SSID='Router' freq=2412 MHz)  
<3>CTRL-EVENT-REGDOM-CHANGE init=COUNTRY_IE type=COUNTRY alpha2=DE  
<3>CTRL-EVENT-DISCONNECTED bssid=b0:f2:08:89:a2:0c reason=1 locally_generated= 1  
<3>CTRL-EVENT-REGDOM-CHANGE init=CORE type= WORLD  
<3>CTRL-EVENT-REGDOM-CHANGE init=CORE type= WORLD  
<3>Trying to associate with b0:f2:08:89:d5:f9 (SSID='Router' freq=2412 MHz)  
<3>Trying to associate with b0:f2:08:89:a2:0c (SSID='Router' freq=2412 MHz)  
<3>Trying to associate with b0:f2:08:89:d5:f9 (SSID='Router' freq=2412 MHz)  
```

</details>

~~Possible fix (thanks [@j-Nimble](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/discussions/178#discussioncomment-10585979))
`sudo apt install -y network-manager net-tools ifupdown dhcpcd5 usbutils`~~ 
Did not fix the issue for me, i had to disable WPA3 support at my router (had WPA2+3 enabled), then i could connect again.
It seems, the issue is that the Ubuntu 20.04.6 LTS / Network-Manager 1.22.10 version on the pad is too old for WPA3 support and the general support for this OS runs out in April 2025.

I tried to upgrade the pad to Ubuntu 22.04.5 LTS version, but after that, it no longer boots up, so i had to re-flash it again to stock.
Maybe someone figures it out, how to upgrade it correctly.

Unfortunately did i stuck then at the next steps with the network-manager, where i have to connect my WiFi via nmcli:
`nmcli device wifi connect "Your_SSID" password "Your_Password"`
It just couldn't connect and i needed to reboot the device to manual enter the WiFi stuff again.
But after everything was finally setup, i just deleted all saved Wifi's via the pad screen and connected again, also via the pad screen.

~~<ins>**2.**</ins> I couldn't install Klipper via kiauh when Python 3.9.19 is installed, i skipped the "sp_installer2.sh" part and just executed the commands in it, which are not change Python stuff.~~

~~Possible fix (thanks [@Hydra20vt](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/discussions/186#discussioncomment-10544699)):
`pip install setuptools==70.3.0`~~
This issue was fixed by @Guilouz on Oct 25, 2024 by updating the sp_installer2.sh.

~~<ins>**3.**</ins> After i completed the install of this [part](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Install-Official-Builds-(1-instance)), the pad no longer boots up, it freezes on the boot logo.~~
The first reboot just took ages, probably because of the broken WiFi setup.

~I did not test the possible fixes yet, i am happy that it runs atm. for me.~

It would be nice if you can address the issues in your guide.

**Comment by Hydra20vt** [comment]
i have done some research during a new installation and the simpliest method to fix Klipper setup problem is 
`pip install --upgrade setuptools[core]`

it just download the latest version of setup tools replacing the bugged one

**Reply by enzoav** [reply]
I am having the hardest time trying to install Klipper via kiauh. When I first ran through the guide, I was getting the virtualenv error and Klipper status was reading Incomplete inside kiauh. 

I ran the pip install --upgrade setuptools[core] command and that got rid of that error. Now I'm getting a new error while trying to install Klipper.
![Screenshot 2024-09-10 at 6 11 10 PM](https://github.com/user-attachments/assets/ed34e5d3-cd41-4a79-a1f7-4e285e9dc4d1)

**Reply by enzoav** [reply]
I went back and fixed my issue by reinstalling kiauh from the kiauh main repo
cd ~ && git clone https://github.com/dw-0/kiauh.git
Then I ran through the kiauh installs of klipper, moonraker, and mainsail like in the guide and then continued the guide. I had to run the klipperscreen shell multiple times ( ./KlipperScreen/scripts/KlipperScreen-install.sh ) until it finally asked the question to install the network manager and i said no as per the guide and finally I was able to contiue and I finally am able to see the v400 again in klipperscreen.  I think the issues are just that things have updated since the guide and its scripts were made and they both may need some minor tweaking.

**Comment by Mingasux** [comment]
2 months on and nothing.  I've tried every "fix" from the Youtube script to every mentioned line in the discussions but none are working.  Gone back to stock and waiting for an actual fix on this @Guilouz

**Reply by enzoav** [reply]
> 2 months on and nothing. I've tried every "fix" from the Youtube script to every mentioned line in the discussions but none are working. Gone back to stock and waiting for an actual fix on this @Guilouz

What issue are you having? I may be able to help.  At least, I will try. I do have it running solid on Vanilla Klipper.

**Comment by Dark-Noir** [comment]
Today i reinstalled my Speeder Pad and updated the [first post](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/discussions/187#discussion-7148388).

**Reply by brdi** [reply]
Greatz that it works again.
I hope I can get by the issues I am experiencing.
Connecting and Wifi were no problem at all.
I am runnning into a prpblem when installing Klipper using kiauh.
There seems to be a pip and/or a python version issue.
Did you have that kind of promlems too?

**Reply by Dark-Noir** [reply]
Sorry, i can't really help anymore, because i sold mine last year.
The last update of this repo was 2 years ago, but dependencies got updated and are probably no longer compatible with it.
Maybe someone can help, but many people, including the dev of this moved on.

**Reply by brdi** [reply]
Ok, i'll get the thing working again . Will just take Some time. But i'm allmost there and Will post an adapted procedure.
Good luck

**Comment by kaktuss77** [comment]
Hi everybody, 

Confirm that is your SSID is WPA2/WPA3, the pad wouldn't connect. I have to connect to other SSID with WPA2 only and now the pad is connected.

Regards,

**Reply by cottec** [reply]
https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/discussions/178#discussioncomment-10448837
